### DISCUSSION TITLE
Faking responses?

### DISCUSSION THREAD

**Discussion by mariomadproductions** [evilsocket/opensnitch#531] [Ideas]
Would it make sense for OpenSnitch sto have the ability to "fake" responses to software (as some software misbehaves if it can't connect to a server)? Or would it make more sense for OpenSnitch to instead have a "redirect" option, where it sends the request to your own server that is designed to imitate the real server? I'm just asking as I was going to open a feature request issue for the former, but I wasn't sure if it really made sense.

Edit: Seems like #437 is a feature request for the latter?

**Comment by gustavo-iniguez-goya** [comment]
Hi @mariomadproductions ,

I think a factible feature would be to put a mark on packets of an application, and apply whatever rule you want to that mark: reject, drop, redirect, etc... Conceptually it seems to be very easy, but I haven't had time to make a PoC yet. It was requested in that issue, yes.

faking responses much more work. It would require to modify packets, and in the scenario you mentioned, knowing how the application works to reply the answer it requires.

> as some software misbehaves if it can't connect to a server

Do you have an example of this problem? If so I'd like to know what app it is. Sometimes we cause apps to crash, like transmission some versions ago.

**Comment by mariomadproductions** [comment]
[RomVault](https://www.romvault.com/) (running with mono) is slow to start if you deny its connections. In this case I don't mind allowing the connections too much, but it may be something that happens with other apps too.

**Comment by gustavo-iniguez-goya** [comment]
ok, I see. Yes, that's also a pretty common problem when blocking ads/malware urls: #481 
I added an option to reject connections instead of dropping them and it certainly helps to reduce web pages loading times, so maybe it also helps in this scenario.

It'll be included in the next RC version that I hope to release soon.
